### ISSUE TITLE
Incorrect exception thrown for authenticating against an SFTP server with invalid password

### ISSUE THREAD

**Issue by mathanpt** [hierynomus/sshj#315]
The following exception is thrown for authenticating against an SFTP server with invalid password and it doesn't seem quite right

`net.schmizz.sshj.userauth.UserAuthException: Exhausted available authentication methods`

**Comment by mathanpt** [comment]
Any help would be greatly appreciated

**Comment by dazed19** [comment]
Hi

It's generally the case that a server shouldn't indicate that a password check has failed, since this would indicate that the account being used is valid  - this holds (I believe) for SSH/SFTP access - see [this](https://security.stackexchange.com/questions/17816/username-and-or-password-invalid-why-do-websites-show-this-kind-of-message-i) - so the less explict response is appropriate in this case.

**Comment by hierynomus** [comment]
@dazed19 You're right for a server, but SSHJ is not a server, but rather a client.

I will have a look to see whether I can improve the exceptions coming out of the authentication. 

This is what regular SSH does:
```
> ssh jeroen@XXXXX
jeroen@XXXXX's password:
Permission denied, please try again.
jeroen@XXXXX's password:
Permission denied, please try again.
jeroen@XXXXX's password:
Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).
```

**Comment by dazed19** [comment]
@hierynomus - I'm happy to be proved wrong, but isn't SSHJ a client connecting to a server - and the server won't (shouldn't) be telling the client what it didn't like about the authentication failure?.  Whoever is in control of the server can determine the failure details from the SSHD logs.

From a client perspective I agree you can add authentication methods (into SSHJ) and report the list used on failure - but without being able to say explicitly what element failed (my expectation).

**Comment by hierynomus** [comment]
@dazed19 You're right, we can't provide more info than the SSH server is telling us. And when looking through the code just now, I think we provide the information the server told us in the caused by stacktrace.
